File: packages/capp-sdk/capp/modules/routing.py
import httpx
from typing import List
from ..models import RouteAnalysisResult, FXRate
from .._utils import handle_api_error

class RoutingModule:
    def __init__(self, client: httpx.AsyncClient):
        self._client = client
        
    async def analyze(
        self, 
        amount: float, 
        from_currency: str, 
        to_currency: str, 
        corridor: str
    ) -> List[RouteAnalysisResult]:
        res = await self._client.post("/routing/analyze", json={
            "amount": amount,
            "from_currency": from_currency,
            "to_currency": to_currency,
            "corridor": corridor
        })
        handle_api_error(res)
        data = res.json()
        # Returning list to match the spec `routes[0].chain`
        return [RouteAnalysisResult(**item) for item in data.get("routes", [])]

    async def get_fx_rate(self, pair: str) -> FXRate:
        res = await self._client.get("/routing/fx", params={"pair": pair})
        handle_api_error(res)
        return FXRate(**res.json())

File: packages/capp-sdk-ts/src/modules/routing.ts
import { CAPPClient } from '../client';
import { RouteAnalysisResult, FXRate } from '../models';
import { handleApiError } from '../utils';

export class RoutingModule {
    constructor(private client: CAPPClient) { }

    public async analyze(params: {
        amount: number;
        fromCurrency: string;
        toCurrency: string;
        corridor: string;
    }): Promise<RouteAnalysisResult[]> {
        const res = await this.client.fetch('/routing/analyze', {
            method: 'POST',
            body: JSON.stringify({
                amount: params.amount,
                from_currency: params.fromCurrency,
                to_currency: params.toCurrency,
                corridor: params.corridor
            })
        });
        await handleApiError(res);
        const data = await res.json();
        return data.routes || [];
    }

    public async getFxRate(pair: string): Promise<FXRate> {
        const res = await this.client.fetch(`/routing/fx?pair=${encodeURIComponent(pair)}`);
        await handleApiError(res);
        return res.json();
    }
}
